proxy/src/com/aerospike/client/proxy/CommandProxy.java

```diff
@@ -49,6 +49,12 @@ public CommandProxy(GrpcCallExecutor executor, Policy policy) {
 	final void execute() {
 		if (policy.totalTimeout > 0) {
 			deadline = System.nanoTime() + TimeUnit.MILLISECONDS.toNanos(policy.totalTimeout);
+		} else if (policy.socketTimeout > 0) {
+			deadline = System.nanoTime() + TimeUnit.MILLISECONDS.toNanos(policy.socketTimeout);
+		} else {
+			// TODO: @BrianNichols call SHOULD expire at some time. What
```

Comment:
All calls SHOULD expire, else there might instances where a call is stuck internally in one of the queues.